cmd/soroban-rpc/internal/test/simulate_transaction_test.go

```diff
@@ -810,4 +783,38 @@ func TestSimulateTransactionBumpAndRestoreFootprint(t *testing.T) {
 	tx, err = txnbuild.NewTransaction(params)
 	assert.NoError(t, err)
 	sendSuccessfulTransaction(t, client, sourceAccount, tx)
+
+	// Wait for expiration again and check the pre-restore field when trying to exec the contract again
+	waitForExpiration()
+
+	simulationResult := simulateTransactionFromTxParams(t, client, invokeIncPresistentEntryParams)
+	assert.NotZero(t, simulationResult.PreRestoreTransactionData)
+	assert.NotZero(t, simulationResult.PreRestoreMinResourceFee)
+
+	params = preflightTransactionParamsLocally(t,
+		txnbuild.TransactionParams{
+			SourceAccount:        &account,
+			IncrementSequenceNum: true,
+			Operations: []txnbuild.Operation{
+				&txnbuild.RestoreFootprint{},
+			},
+			Preconditions: txnbuild.Preconditions{
+				TimeBounds: txnbuild.NewInfiniteTimeout(),
+			},
+		},
+		methods.SimulateTransactionResponse{
+			TransactionData: simulationResult.PreRestoreTransactionData,
+			MinResourceFee:  simulationResult.PreRestoreMinResourceFee,
+		},
+	)
+	tx, err = txnbuild.NewTransaction(params)
+	assert.NoError(t, err)
+	sendSuccessfulTransaction(t, client, sourceAccount, tx)
+
+	// Finally, we should be able to send the inc host function invocation now that we
+	// have pre-restored the entries
+	params = preflightTransactionParamsLocally(t, invokeIncPresistentEntryParams, simulationResult)
+	tx, err = txnbuild.NewTransaction(params)
+	assert.NoError(t, err)
+	sendSuccessfulTransaction(t, client, sourceAccount, tx)
 }
```

Comment:
@Shaptic here's an example of how to use it.

Reply:
Thank you! This means the flow is something like:

```
result = simulateTransaction(invoke)
if result.PreRestoreTransactionData != undefined: 
    submitTransaction(restore, transactionData=result.PreRestoreTransactionData)
    result = simulateTransaction(invoke)
submitTransaction(invoke, transactionData=result.transactionData)
```

Correct?

Reply:
Yes, assuming you somehow also use the minfee too.

Reply:
Actually, you don't need the second simulation

```
result = simulateTransaction(invoke)
if result.PreRestoreTransactionData != undefined: 
    submitTransaction(restore, transactionData=result.PreRestoreTransactionData)
submitTransaction(invoke, transactionData=result.transactionData)
```

Reply:
@Shaptic did you see my udpate above? You don't need to re-run the transaction simulation